pkg/services/sqlstore/migrations/ualert/alert_rule_version_guid_mig.go

```diff
@@ -58,26 +62,59 @@ func (c setRuleGuidMigration) SQL(migrator.Dialect) string {
 }
 
 func (c setRuleGuidMigration) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
-	var results []string
-	if err := sess.SQL("SELECT uid FROM alert_rule").Find(&results); err != nil {
-		return err
-	}
-	if len(results) == 0 {
-		mg.Logger.Debug("no rules found")
-		return nil
-	}
-	for _, uid := range results {
-		u := uuid.NewString()
-		_, err := sess.Exec("UPDATE alert_rule_version SET rule_guid = ? WHERE rule_uid = ?", u, uid)
-		if err != nil {
-			mg.Logger.Error("Failed to update alert_rule_version table", "error", err)
+	var lastId *int64
+	for {
+		var results []int64
+		qq := sess.Table(`alert_rule`).Select("id").OrderBy("id").Limit(500)
+		if lastId != nil {
+			qq = qq.Where("id > ?", lastId)
+		}
+		if err := qq.Find(&results); err != nil {
 			return err
 		}
-		_, err = sess.Exec("UPDATE alert_rule SET guid = ? WHERE uid = ?", u, uid)
+		if len(results) == 0 {
+			break
+		}
+		bd := strings.Builder{}
+		for idx, id := range results {
+			u := uuid.NewString()
```

Comment:
In the case where we're using the MySQL dialect, I wonder if using the SQL `UUID()` function instead would be more performant, thus offloading the generation to the DB?

Reply:
I tried, this function cannot be used in the column's default